internal/observability/call.go

```diff
@@ -75,69 +75,67 @@ type call struct {
 	levels *levels
 }
 
+type callResult struct {
+	err            error
+	ctxOverrideErr error
+
+	isApplicationError   bool
+	applicationErrorMeta *transport.ApplicationErrorMeta
+
+	requestSize  int
+	responseSize int
+}
+
 type levels struct {
 	success, failure, applicationError zapcore.Level
 }
 
-func (c call) End(err error, requestSize int) {
-	c.endWithAppError(err, false /* isApplicationError */, nil /* applicationErrorMeta */, requestSize, 0)
+func (c call) End(res *callResult) {
```

Comment:
let's give change this to have `res callResult` -> this is less expensive in golang (GC)